lmcache/cache_engine.py

```diff
@@ -301,17 +313,30 @@ def retrieve(
                 
                 For vllm, it should have the shape of 
                 [num_tokens, num_heads, head_size]
+                
+            mask: a boolean mask of tokens indicating which tokens'
+            KV Cache should be retrieved
 
         Output:
             kv_tensors: the kv cache of the tokens, in the format of nested 
                         tuples. Will be an empty tuple if no kv cache is 
                         retrieved.
 
-            num_tokens: the number of tokens in the kv cache
+            ret_mask: indicate which tokens are retrieved
         """
+        chunked_skip_len = 0
```

Comment:
The changes here seem to conflict with #153. Did you pulled/picked some old codes from there?

Reply:
Another side note. @KuntaiDu and I have talked about these examples. They contain too much repetitive code. I'll refactor them later.

Reply:
There are new commits after I wrote this PR. I think simply keeping the code in #153 should work.